src/main/kotlin/no/item/xp/plugin/GenerateTypeScriptTask.kt

```diff
@@ -1,27 +1,96 @@
 package no.item.xp.plugin
 
+import arrow.core.Either
 import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
 import javax.inject.Inject
+import javax.xml.parsers.DocumentBuilderFactory
+import no.item.xp.plugin.models.GeneratedField
+import no.item.xp.plugin.models.XmlFile
+import no.item.xp.plugin.models.XmlType
+import no.item.xp.plugin.parser.parse
+import no.item.xp.plugin.util.GenerateInterfaceName
 import org.gradle.api.DefaultTask
 import org.gradle.api.tasks.TaskAction
+import org.w3c.dom.Document
+import java.lang.Exception
+import javax.xml.parsers.DocumentBuilder
 
+@Suppress("UNUSED_VARIABLE")
 open class GenerateTypeScriptTask @Inject constructor(private val extension: GenerateTypeScriptExtension) : DefaultTask() {
 
+  private val docBuilder: DocumentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+
     @TaskAction
     fun generateTypeScript() {
-      val files = findXmlFiles()
-      println("Found XML files: ")
-      println("--------------------")
-      files.forEach { println(it) }
-      // 2.send dem til en parser
-      // 3.send resultatet til en writer
-      // 4.exit
-    }
-
+      val files: Sequence<String> = findXmlFiles()
+      files.forEach { parseFile(it) }
+  }
   private fun findXmlFiles(): Sequence<String> {
     return File(extension.fileDir)
       .walk()
-      .filter { it.isFile && it.extension == "xml" }
+      .filter { it.isFile }
       .map { it.absolutePath }
   }
+
+  private fun parseFile(filePath: String) {
+    val xmlType: Either<Throwable, XmlType> = getXmlType(filePath)
+    var xmlTypeValue: XmlType? = null
+    when(xmlType){
+      is Either.Right -> {
+        xmlTypeValue = xmlType.b
+      }
+    }
+    if(xmlTypeValue!=null) {
+      val xmlFile: XmlFile = generateXmlFile(filePath, xmlTypeValue)
+      val document: Document = readXml(xmlFile)
+      parse(document)
+
+      val interfaceName: String? = GenerateInterfaceName(File(filePath))
+      val xml: Either<Throwable, Sequence<GeneratedField>> = parse(document)
+      /*if (interfaceName != null) {
+      TypeScriptWriter(interfaceName, xmlType, xml)
+    }*/
+    }
+  }
+
+  private fun getXmlType(filePath: String): Either<Throwable, XmlType> {
+    val child: Path = Paths.get(filePath).toAbsolutePath()
+    return returnIsChild(child)
+  }
+
+  private fun returnIsChild(child: Path): Either<Throwable, XmlType> {
+    for (pair: Pair<String, XmlType> in pairList) {
+      if (isChild(child, pair.first)) {
+        return Either.right(pair.second)
+      }
+    }
+    return Either.left(Exception("could not find XmlType for $child"))
+  }
+
+  private val pairList = listOf(
```

Comment:
This should have a more descriptive name based on what data it contains. And it's a constant, so the name should be upper case.
```suggestion
  private val DIRECTORIES_AND_XMLTYPE = listOf(
```

Reply:
You decided to go for a tuple here (`Pair`), and that is completely fine. A tuple is like a `data class`, but you don't have to name the fields (we only care about the type). Just make sure that it is easy enough to understand what you are trying to acheve.
(this is just a tip, you don't have to do any changes here)